### PR TITLE
Sulyi

### PR DESCRIPTION
**Pull request by sulyi**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

**Greetings**,
I've made some changes in the *utils.js_to_json* function when dealing with comments. I've added a *test_js_to_json_landofoz* testcase for it. Yet, It could be be a bit cleaner  if whitespaces were ignored by the assertion.

Similarly in the *jsinterp* module, I've started to tackle handling comments, but I'm less sure about that one. And I've added an extract_arguments method that uses "context-free grammar".

I've also added a *youwatch* extractor, but I'm not sure if it fits the copyright-protection policy.

I try to keep these features on separate branches, and also mark them in  my comments. I keep another branch for my merges.

I'm open to suggestions or remarks, and wouldn't mind some discussion about the underlining maths of these problems. I'm looking forward to work together.

Cheers,
axx

### CODE REVIEW

File: youtube_dl/utils.py [hunk 1]
```diff
@@ -2109,10 +2109,13 @@ def fix_kv(m):
...
         return '"%s"' % v
 
+    # fixing , followed nothing, but comments
+    # fixing unnecessary ? in /\*.*?\*/
+    # fixing greedy comment
     return re.sub(r'''(?sx)
         "(?:[^"\\]*(?:\\\\|\\['"nurtbfx/\n]))*[^"\\]*"|
         '(?:[^'\\]*(?:\\\\|\\['"nurtbfx/\n]))*[^'\\]*'|
-        /\*.*?\*/|,(?=\s*[\]}])|
+        /\*[^*]*\*/|,(?=(\s|(/\*[^*]*\*/))*[\]}])|
```
Comment (fstirlitz) [inline youtube_dl/utils.py]:
```
/**********/
```

**Comment by dstftw** [comment]
This intersects with #11261. Remove all intersecting parts and open PR per feature. Also make proper PR titles.

**Comment by sulyi** [comment]
Of course, #11261 meant to fix my messed up branching.

**Comment by yan12125** [comment]
@sulyi Could you close this and other superseded pull requests?

File: youtube_dl/utils.py (hunk 1 above)
Comment (sulyi) [inline youtube_dl/utils.py]: Yes,that's been fixed, but now /*** /* */ ***/ is accepted yet it shouldn't be.

**Comment by sulyi** [comment]
I don't know, can I?

**Comment by yan12125** [comment]
There's a "Close pull request" button below

**Comment by sulyi** [comment]
Thanks.

**Pull request closed without merging**